src/components/CustomAppDetails/Versions.js

```diff
@@ -0,0 +1,271 @@
+    SingleSelectOption,
+    Button,
+    Table,
+    TableHead,
+    TableRowHead,
+    TableCellHead,
+    TableBody,
+    TableRow,
+    TableCell,
+} from '@dhis2/ui'
+import moment from 'moment'
+import React, { useState } from 'react'
+import semver from 'semver'
+import { useApi } from '../../api'
+import { channelToDisplayName } from './channel-to-display-name'
+import styles from './CustomAppDetails.module.css'
+
+const ChannelCheckbox = ({
+    name,
+    label,
+    channelsFilter,
+    setChannelsFilter,
+}) => {
+    const handleChange = ({ checked }) => {
+        const newState = new Set(channelsFilter)
+        if (checked) {
+            newState.add(name)
+        } else {
+            newState.delete(name)
+        }
+        setChannelsFilter(newState)
+    }
+
+    return (
+        <div className={styles.channelCheckbox}>
+            <Checkbox
+                checked={channelsFilter.has(name)}
+                disabled={channelsFilter.size === 1 && channelsFilter.has(name)}
+                onChange={handleChange}
+                label={label}
+            />
+        </div>
+    )
+}
+
+ChannelCheckbox.propTypes = {
+    channelsFilter: PropTypes.object.isRequired,
+    label: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    setChannelsFilter: PropTypes.func.isRequired,
+}
+
+const Filters = ({
+    versions,
+    channelsFilter,
+    setChannelsFilter,
+    dhisVersionFilter,
+    setDhisVersionFilter,
+    dhisVersions,
+}) => {
+    const hasChannel = channel => versions.some(v => v.channel === channel)
+
+    return (
+        <div className={styles.versionsFilters}>
+            <h3 className={styles.sectionSubheader}>
+                {i18n.t('Channel', { context: 'AppHub release channel' })}
+            </h3>
+            {Object.keys(channelToDisplayName)
+                .filter(hasChannel)
+                .map(name => (
+                    <ChannelCheckbox
+                        key={name}
+                        name={name}
+                        label={channelToDisplayName[name]}
+                        channelsFilter={channelsFilter}
+                        setChannelsFilter={setChannelsFilter}
+                    />
+                ))}
+
+            <div className={styles.dhisVersionSelect}>
+                <SingleSelectField
+                    placeholder={i18n.t('Select a version')}
+                    label={i18n.t('Compatible with DHIS2 version')}
+                    clearable
+                    selected={dhisVersionFilter}
+                    onChange={({ selected }) => setDhisVersionFilter(selected)}
+                >
+                    {dhisVersions.map(dhisVersion => (
+                        <SingleSelectOption
+                            key={dhisVersion}
+                            label={dhisVersion}
+                            value={dhisVersion}
+                        />
+                    ))}
+                </SingleSelectField>
```

Comment:
This doesn't make a lot of sense here, as only versions compatible with the current server version should be available.  It might be nice to show "disabled" versions which are incompatible, but the select box doesn't make much sense.

Also, showing the current and two previous versions is incorrect.  I will comment separately there.